src/Nest/CommonAbstractions/Extensions/Extensions.cs

```diff
@@ -169,8 +188,8 @@ internal static void ForEach<T>(this IEnumerable<T> enumerable, Action<T> handle
 
 		internal static List<T> ToListOrNullIfEmpty<T>(this IEnumerable<T> enumerable)
 		{
-			var list = enumerable?.ToList();
-			return list.HasAny() ? list : null;
+			var list = enumerable.AsInstanceOrToListOrNull();
+			return list == null || list.Count > 0 ? list : null;

```

Comment:
actually it is correct :) if `list == null || list.Count > 0` = `list != null && list.Count > 0` has the same effect here. Changing it to the latter for readability